gtfu/src/web/graas-report-web.js

```diff
@@ -0,0 +1,343 @@
+            if(searchY >= mapScrollTop){
+                searchY -=headerHeight;
+                searchObject = trips[i].map;
+
+            // If the click occurs within the header, adjust y value and search timeline objects
+            } else {
+                searchY -=timelineScrollTop;
+                searchObject = trips[i].timeline;
+            }
+            if (objectContainsPoint(searchObject, event.pageX, searchY)){
+                drawMetadata(trips[i]);
+                selectTrip(mapCtx, trips[i].map);
+                selectTrip(timelineCtx, trips[i].timeline);
+                scrollToTrip(trips[i])
+                break;
+            }
+        }
+    });
+}
+
+function objectContainsPoint(object, x, y){
+    if(x >= object.x){
+        if(x <= (object.x + object.width)){
+            if(y >= object.y ){
+                if(y <= (object.y + object.height)){
+                    return true;
+                }
+            }
+        }
+    }
+}
+
+function drawReport(){
+    console.log("Drawing map...");
+    timelineCtx.drawImage(img, 0, 0, imageWidth, timelineHeight, 0, 0, canvasWidth, timelineCanvasHeight);
+    mapCtx.drawImage(img, 0, timelineHeight, imageWidth, mapHeight, 0, 0, canvasWidth, mapCanvasHeight);
+}
+
+function drawMetadata(trip){
+    console.log("drawTooltip()");
+
+    var margin = 5;
+
+    metadataWidth = trip.map.width;
+    metadataX = trip.map.x + metadataWidth;
+    metadataY = trip.map.y;
+    metadataHeight = trip.map.height;
+
+    while ((tooltipItems.length * (font_size + margin) >= metadataHeight)
+            || (getMaxTextWidth(trip) + margin * 2 >= metadataWidth))  {
+        console.log("reducing font size...");
+        font_size -= 5;
+    }
+
+    // Metadata on right side unless it would go over edge
+    if((metadataX + metadataWidth) > canvasWidth){
+        metadataX -= (2 * metadataWidth);
+    }
+
+    // Draw tooltip box outline
+    mapCtx.beginPath();
+    mapCtx.lineWidth = 1
+    mapCtx.strokeStyle = "black";
+    mapCtx.fillStyle = "white";
+    mapCtx.fillRect(metadataX, metadataY, metadataWidth, metadataHeight);
+    mapCtx.rect(metadataX, metadataY, metadataWidth, metadataHeight);
+
+    // Draw tooltip text
+    mapCtx.stroke();
+    mapCtx.fillStyle = "black";
+    mapCtx.font= `${font_size}px ${font}`;
+
+    for (var i = 0; i < tooltipItems.length; i++) {
+        metadataY += (font_size + margin);
+        mapCtx.fillText(`${tooltipItems[i]}: ${trip[tooltipItems[i]]}`, metadataX + margin , metadataY);
+    }
+}
+
+function getMaxTextWidth(trip) {
+    maxWidth = 0;
+    mapCtx.font= font_size + "px " + font;
+    for (var i = 0; i < tooltipItems.length; i++) {
+        text = tooltipItems[i] + ": " + trip[tooltipItems[i]];
+        length = mapCtx.measureText(text).width;
+        if (length > maxWidth){
+            maxWidth = length;
+        }
+    }
+    return Math.round(maxWidth);
```

Comment:
It's pedantery, but I think you want Math.ceil() here

Reply:
Since ceil() rounds down and round() rounds up, my understanding is that ceil() would result in the text width getting closer to the edge of the box. And round() would result in a bit more of a buffer.